include/seqan3/alignment/pairwise/detail/pairwise_alignment_algorithm_banded.hpp

```diff
@@ -0,0 +1,310 @@
+     */
+    pairwise_alignment_algorithm_banded(alignment_configuration_t const & config) : base_algorithm_t(config)
+    {
+        using seqan3::get;
+
+        auto band = get<seqan3::align_cfg::band_fixed_size>(config);
+
+        lower_diagonal = band.lower_diagonal.get();
+        upper_diagonal = band.upper_diagonal.get();
+
+        if (upper_diagonal < lower_diagonal || upper_diagonal < 0 || lower_diagonal > 0)
+            throw invalid_alignment_configuration{to_string("The selected band [",
+                                                            lower_diagonal,
+                                                            ":",
+                                                            upper_diagonal,
+                                                            "] is not supported.")};
+    }
+    //!\}
+
+    /*!\name Invocation
+     * \{
+     */
+    //!\copydoc seqan3::detail::pairwise_alignment_algorithm::operator()(indexed_sequence_pairs_t && indexed_sequence_pairs, callback_t && callback)
+    template <indexed_sequence_pair_range indexed_sequence_pairs_t, typename callback_t>
+    //!\cond
+        requires std::invocable<callback_t, alignment_result_type>
+    //!\endcond
+    void operator()(indexed_sequence_pairs_t && indexed_sequence_pairs, callback_t && callback)
+    {
+        using result_value_t = typename alignment_result_value_type_accessor<alignment_result_type>::type;
+        using std::get;
+
+        for (auto && [sequence_pair, idx] : indexed_sequence_pairs)
+        {
+            result_value_t res{};
+            res.id = idx;
+
+            res.score = compute_matrix(get<0>(sequence_pair), get<1>(sequence_pair));
+            callback(alignment_result_type{res});
+        }
+    }
+    //!\}
+
+protected:
+    /*!\brief Checks wether the band given the sequence sizes is valid
+     *
+     * \param[in] sequence1_size The size of the first sequence.
+     * \param[in] sequence2_size The size of the second sequence.
+     *
+     * \throws seqan3::invalid_alignment_configuration if the band is invalid for the given sequence sizes and the
+     *         alignment configuration.
+     */
+    void check_valid_band_configuration(size_t const sequence1_size, size_t const sequence2_size) const
+    {
+        bool const upper_diagonal_ends_before_last_cell = (upper_diagonal + sequence2_size) < sequence1_size;
+        bool const lower_diagonal_ends_behind_last_cell = (-lower_diagonal + sequence1_size) < sequence2_size;
+
+        if (upper_diagonal_ends_before_last_cell || lower_diagonal_ends_behind_last_cell)
+            throw invalid_alignment_configuration{to_string("The selected band [",
+                                                            lower_diagonal,
+                                                            ":",
+                                                            upper_diagonal,
+                                                            "] does not cover the last cell.")};
```

Comment:
```suggestion
            throw invalid_alignment_configuration{"The selected band [" + std::to_string(lower_diagonal) + ":" + std::to_string(upper_diagonal) + "] does not cover the last cell.")};
```

Reply:
I thought we wanted to allow these? And return `std::nullopt` score, if the tracked cell is not within band?

Reply:
That might be true but is nowhere covered yet by the alignment or alignment result. So that would be a different issue.